python/ml4ir/model/optimizer.py

```diff
@@ -3,6 +3,10 @@
 from tensorflow.keras.optimizers.schedules import ExponentialDecay
 
 
+# Constants
+CLIP_VALUE = 5.0
+
+
 def get_optimizer(
```

Comment:
pass in `clipvalue=CLIP_VALUE` here, and use it for all inside?

Reply:
Wait. I think you are looking at outdated code. I've fixed this with reading clipvalue as an arg.

Reply:
ok I'm walking through the PR in full right now!

Reply:
Cool. I will stop replying for a while.